File: solutions/typescript/secret-handshake/1/secret-handshake.ts
const Code = {
  WINK: 0b00001,
  DOUBLE_BLINK: 0b00010,
  CLOSE_EYES: 0b00100,
  JUMP: 0b01000,
  REVERSE: 0b10000,
}

const codes = [Code.WINK, Code.DOUBLE_BLINK, Code.CLOSE_EYES, Code.JUMP];

export function commands(num: number): string[] {
  const command: string[] = [];
  for (const code of codes) {
    if ((code & num) !== 0)
      command.push(getMessageCode(code));
  }
  if ((num & Code.REVERSE) !== 0)
    command.reverse();
  return command;
}

function getMessageCode(code: number): string {
  switch (code) {
    case Code.WINK:
      return "wink";
    case Code.DOUBLE_BLINK:
      return "double blink";
    case Code.CLOSE_EYES:
      return "close your eyes";
    case Code.JUMP:
      return "jump";
    default:
      throw new Error(`Invalid code number: ${code}`);
  }
}

File: solutions/csharp/secret-handshake/2/SecretHandshake.cs
using System.Collections.Generic;

public static class SecretHandshake
{
    public static string[] Commands(int commandValue)
    {
        List<string> commands = new();
        if ((commandValue & 0b_0001) != 0)
            commands.Add("wink");
        if ((commandValue & 0b_0010) != 0)
            commands.Add("double blink");
        if ((commandValue & 0b_0100) != 0)
            commands.Add("close your eyes");
        if ((commandValue & 0b_1000) != 0)
            commands.Add("jump");
        if ((commandValue & 0b_0001_0000) != 0)
            commands.Reverse();
        return commands.ToArray();
    }
}
